test/blackbox-tests/test-cases/enabled_if/eif-public-name-collision.t

```diff
@@ -0,0 +1,65 @@
+Show that public library names can't be defined twice, even in different
+contexts
```

Comment:
It's worth adding a statement on if this is a bug or an intentional limitation.

Reply:
to be honest, we're not sure. It depends a bit on the usage, and how they're installed:

- for Melange libraries that clash, it's not a problem if they have the same public name, as we install melange artifacts under `<lib_name>/melange` anyway. 
- for other libraries, if we decided to install them in the same context, they would be overwritten.

We're working on an installation RFC with the aim of making a special case for a "melange context" that can be installed in the same prefix as native libraries with the same name.

Reply:
Okay, well I think this particular issue is easily fixed. We just ignore libraries with `enabled_if` for the purposes of `libname_conflict_map`.